### ISSUE TITLE
Problems with images 

### ISSUE THREAD

**Issue by hustonU** [DotSpatial/DotSpatial#889]
Hi everybody!

I have pictures that are made by UAV.
I`m using GDAL to project its by GCPsToGeoTransform.
After that I want to display its. I create copy of native jpeg file (30 Mb) to another image but when I using
- tif format - wrong colors:
- jpeg - require to much RAM;
- jp2 - wright project and colors, but when image is too much rotate it displays wrong

**Comment by mogikanin** [comment]
so, it is problem with GDAL or DotSpatial?

**Comment by hustonU** [comment]
DotSpatial! Because ArcMap open that without any problems.

**Comment by mogikanin** [comment]
Ok, please attach then screenshots with correct images from ArcMap. 
Also, as I understood, jpeg shows correctly, but takes too much RAM?

**Comment by hustonU** [comment]
I manipulated with GdalRasterProvider and GdalImageProvider but all without decision. Jpeg displays wrigth but without projection or with wrigth projection but with wrong quality!
![screenarcmap](https://cloud.githubusercontent.com/assets/23572442/20455274/aff8e598-ae60-11e6-888c-cf83b0c45f11.png)

**Comment by hustonU** [comment]
ArcMap using pyramids in *.ovr file. Why DotSpatial using mwi and mwm? I tried to create PyramidImage but its also display wrong!

**Comment by hustonU** [comment]
Добрый вечер Максим!

Это пользователь hustonU из github-а!
Знаете ли Вы какое-нибудь решение этого вопроса?

С Уважением Олег!

**Comment by mogikanin** [comment]
Олег, нужно разбираться. Так сразу сложно сказать.

**Comment by hustonU** [comment]
Максим! Еще раз здравствуйте! 
Вы не подскажете схему считывания пирамид растра! К чему привязываться для выбора номера пирамиды? От масштаба?
Буду очень благодарен за ответ!

Воскресенье, 20 ноября 2016, 9:16 +03:00 от Max Miroshnikov <notifications@github.com>:

**Comment by mogikanin** [comment]
Yes, it depends from scale. Take a look into [PyramidImage class](https://github.com/DotSpatial/DotSpatial/blob/master/Source/DotSpatial.Data/PyramidImage.cs#L99)

Also, I confirm that showing this little 5MB jpg takes a lot of RAM. We need investigate and fix it.
![image](https://cloud.githubusercontent.com/assets/840681/20476202/cb0440d6-afe0-11e6-8c42-1d421dc563cc.png)

P.S. Oleg, please use English for comments at github, since other developers don't know Russian :) But feel free to contact me directly via email, you can find my email inside [Contributors file](https://github.com/DotSpatial/DotSpatial/blob/master/Contributors).

**Comment by mogikanin** [comment]
> ArcMap using pyramids in *.ovr file. Why DotSpatial using mwi and mwm? 

Yes, DotSpatial uses own format for pyramids. It's also described inside PyramidImage.cs I guess `mwi` stands for "Map Window Image" of something similar.

**Comment by hustonU** [comment]
Ok! I'm understood! In this way maybe it has a sense to use gdal *.ovr file to create and display pyramids


Понедельник, 21 ноября 2016, 11:55 +03:00 от Max Miroshnikov <notifications@github.com>:

**Comment by mogikanin** [comment]
Ok, regarding using a lot of RAM. This happens because of nature of `InRamImageData` class, which used by default for images. `InRamImageData` stores entire image as raw bitmap in memory, so it is not really suitable for displaying images like this.

Another way is use `GdalImageProvider` and it's `GdalImage` class. This uses not much memory, because it doesn't store entire image as bitmap, and only reads it from hard-drive on demand. But unfortunately,  `GdalImage` is a bit buggy, and image sporadically disappears when you do zoom\pan.

Also, we could implement much more smarter version of  `ImageData` inside `DotSpatial.Data`, without storing entire image as raw bitmap.
